// @fluentui/react-button/lib-commonjs/components/CompoundButton/useCompoundButtonStyles.styles.raw.headless.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    compoundButtonClassNames: function() {
        return compoundButtonClassNames;
    },
    useCompoundButtonStyles_unstable: function() {
        return useCompoundButtonStyles_unstable;
    }
});
const compoundButtonClassNames = {
    root: 'fui-CompoundButton',
    icon: 'fui-CompoundButton__icon',
    contentContainer: 'fui-CompoundButton__contentContainer',
    secondaryContent: 'fui-CompoundButton__secondaryContent'
};
const useCompoundButtonStyles_unstable = (state)=>{
    'use no memo';
    const { appearance, disabled, disabledFocusable, icon, iconOnly, iconPosition, shape, size } = state;
    state.root.className = [
        compoundButtonClassNames.root,
        // Appearance
        appearance && `${compoundButtonClassNames.root}--${appearance}`,
        // Size
        size && `${compoundButtonClassNames.root}--${size}`,
        // Shape
        shape && `${compoundButtonClassNames.root}--${shape}`,
        // Disabled styles
        disabled && `${compoundButtonClassNames.root}--disabled`,
        disabledFocusable && `${compoundButtonClassNames.root}--disabledFocusable`,
        // Icon styles
        icon && iconPosition === 'before' && `${compoundButtonClassNames.root}--iconBefore`,
        icon && iconPosition === 'after' && `${compoundButtonClassNames.root}--iconAfter`,
        icon && iconOnly && `${compoundButtonClassNames.root}--iconOnly`,
        // User provided class name
        state.root.className
    ].filter(Boolean).join(' ');
    if (state.icon) {
        state.icon.className = [
            compoundButtonClassNames.icon,
            state.icon.className
        ].filter(Boolean).join(' ');
    }
    state.contentContainer.className = [
        compoundButtonClassNames.contentContainer,
        state.contentContainer.className
    ].filter(Boolean).join(' ');
    if (state.secondaryContent) {
        state.secondaryContent.className = [
            compoundButtonClassNames.secondaryContent,
            state.secondaryContent.className
        ].filter(Boolean).join(' ');
    }
    return state;
};
